mmdet/apis/ote/apis/detection/openvino_task.py

```diff
@@ -203,11 +143,16 @@ def hparams(self):
 
     def load_inferencer(self) -> OpenVINODetectionInferencer:
         labels = self.task_environment.label_schema.get_labels(include_empty=False)
-        self.confidence_threshold = np.frombuffer(self.model.get_data("confidence_threshold"), dtype=np.float32)[0]
-        return OpenVINODetectionInferencer(labels,
+        _hparams = self.hparams
+        try:
+            self.confidence_threshold = np.frombuffer(self.model.get_data("confidence_threshold"), dtype=np.float32)[0]
+            _hparams.inference_parameters.postprocessing.confidence_threshold = self.confidence_threshold
```

Comment:
I meant that task environment (and hyper parameters as a part of it) should be immutable from with the task. Although this editing might be technically allowed it potentially has some side effects for the outside code. This being said, due to some specifics of implementation this might be not a problem. I suggest to do a copy at least as a workaround.

Reply:
Ok, using deepcopy